AutomatedTesting/Gem/PythonTests/atom_renderer/atom_hydra_scripts/hydra_AtomEditorComponents_DecalAdded.py

```diff
@@ -0,0 +1,143 @@
+    exit_game_mode =            ("Exited game mode",                             "Couldn't exit game mode")
+    is_visible =                ("Entity is visible",                            "Entity was not visible")
+    is_hidden =                 ("Entity is hidden",                             "Entity was not hidden")
+    entity_deleted =            ("Entity deleted",                               "Entity was not deleted")
+    deletion_undo =             ("UNDO deletion success",                        "UNDO deletion failed")
+    deletion_redo =             ("REDO deletion success",                        "REDO deletion failed")
+    no_error_occurred =         ("No errors detected",                           "Errors were detected")
+# fmt: on
+
+
+def AtomEditorComponents_Decal_AddedToEntity():
+    """
+    Summary:
+    Tests the Decal component can be added to an entity and has the expected functionality.
+
+    Test setup:
+    - Wait for Editor idle loop.
+    - Open the "Base" level.
+
+    Expected Behavior:
+    The component can be added, used in game mode, hidden/shown, deleted, and has accurate required components.
+    Creation and deletion undo/redo should also work.
+
+    Test Steps:
+    1) Create a Decal entity with no components.
+    2) Add Decal component to Decal entity.
+    3) UNDO the entity creation and component addition.
+    4) REDO the entity creation and component addition.
+    5) Enter/Exit game mode.
+    6) Test IsHidden.
+    7) Test IsVisible.
+    8) Set Material property on Decal component.
+    9) Delete Decal entity.
+    10) UNDO deletion.
+    11) REDO deletion.
+    12) Look for errors.
+
+    :return: None
+    """
+    import os
+
+    import azlmbr.asset as asset
+    import azlmbr.bus as bus
+    import azlmbr.editor as editor
+    import azlmbr.legacy.general as general
+    import azlmbr.math as math
+
+    from editor_python_test_tools.editor_entity_utils import EditorEntity
+    from editor_python_test_tools.utils import Report, Tracer, TestHelper as helper
+
+    with Tracer() as error_tracer:
+        # Test setup begins.
+        # Setup: Wait for Editor idle loop before executing Python hydra scripts then open "Base" level.
+        helper.init_idle()
+        helper.open_level("Physics", "Base")
+
+        # Test steps begin.
+        # 1. Create a Decal entity with no components.
+        decal_name = "Decal (Atom)"
+        decal_entity = EditorEntity.create_editor_entity_at(math.Vector3(512.0, 512.0, 34.0), f"{decal_name}")
+        Report.critical_result(Tests.decal_creation, decal_entity.exists())
+
+        # 2. Add Decal component to Decal entity.
+        decal_component = decal_entity.add_component(decal_name)
+        Report.critical_result(Tests.decal_component, decal_entity.has_component(decal_name))
+
+        # 3. UNDO the entity creation and component addition.
+        # Requires 3 UNDO calls to remove the Entity completely.
+        for x in range(4):
```

Comment:
This does 4 undos, not 3. Either the comment is wrong or you can do 3 UNDOs.
More information on what each undo does would be super useful